query.go

```diff
@@ -1199,20 +1199,31 @@ func (s *ChainService) GetCFilter(blockHash chainhash.Hash,
 	}()
 
 	var ok bool
-	select {
+	var resultFilter *gcs.Filter
 
-	// We'll return immediately to the caller when the filter arrives.
-	case filter, ok = <-query.filterChan:
-		if !ok {
+	// We will wait for the query to finish before we return the requested
+	// filter to the caller.
+	for {
+		select {
+
+		case filter, ok = <-query.filterChan:
+			if !ok {
+
+				// query has finished, if we have a result we'll return it.
+				if resultFilter == nil {
+					// TODO(halseth): return error?
+					return nil, nil
+				}
+				return resultFilter, nil
```

Comment:
Nit: can just do `return resultFilter, nil`, no need for the `if` above.

Reply:
Fixed.